modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/IgniteSnapshotManager.java

```diff
@@ -822,14 +822,17 @@ private IgniteInternalFuture<SnapshotOperationResponse> initLocalSnapshotStartSt
 
                 snpDir.mkdirs();
 
+                SnapshotFutureTaskResult res = (SnapshotFutureTaskResult)fut.result();
+
                 SnapshotMetadata meta = new SnapshotMetadata(req.requestId(),
                     req.snapshotName(),
                     cctx.localNode().consistentId().toString(),
                     pdsSettings.folderName(),
                     cctx.gridConfig().getDataStorageConfiguration().getPageSize(),
                     grpIds,
                     blts,
-                    (Set<GroupPartitionId>)fut.result(),
+                    res.parts(),
+                    res.snapshotPointer() == null ? null : res.snapshotPointer().index() - 1,
```

Comment:
The logic with `seg - 1` is not clear for me. Can you, please, explain.

Reply:
Remove segment rollover. Store full WALRecord instead